scripts/nightly/get_version.py

```diff
@@ -0,0 +1,31 @@
+"""
+This script is called by the nightly build in Circle CI.
+
+It alters the version of Wagtail to include the build date
+"""
+
+import datetime
+
+from wagtail import VERSION
+
+
+INIT_TEMPLATE = """
+from wagtail.utils.version import get_semver_version, get_version
+
+# major.minor.patch.release.number
+# release must be one of alpha, beta, rc, or final
+VERSION = ({major}, {minor}, {patch}, 'dev', '{datestamp}')
```

Comment:
Don't we need to update wagtail.utils.version to recognise `dev` as a valid release type? https://github.com/wagtail/wagtail/blob/master/wagtail/utils/version.py#L16

Reply:
Any objections to use `alpha` or `beta` instead?

Reply:
Also I'm not sure about `datestamp`. Are we following [semver](https://semver.org/) here?

Reply:
There is no consensus for that.
See https://github.com/semver/semver/issues/200 and newer, an RFC https://github.com/semver/semver/pull/519

Reply:
`'dev'` is the PEP440 standard name to use for a "Developmental release":

> Some projects make regular developmental releases, and system packagers (especially for Linux distributions) may wish to create early releases directly from source control which do not conflict with later project releases.

https://www.python.org/dev/peps/pep-0440/#developmental-releases

I think this description best fits with nightly releases.

> Also I'm not sure about datestamp. Are we following semver here?

I'm following [PEP 440](https://www.python.org/dev/peps/pep-0440) which I think is most appropriate for python packages.

Reply:
[Release documentation](http://docs.wagtail.io/en/v2.5.1/contributing/release_process.html#release-cadence) says that Wagtail uses "loose form" of SemVer:

> Wagtail uses a loose form of [semantic versioning](http://semver.org/). SemVer makes it easier to see at a glance how compatible releases are with each other. It also helps to anticipate when compatibility shims will be removed. It’s not a pure form of SemVer as each feature release will continue to have a few documented backwards incompatibilities where a deprecation path isn’t possible or not worth the cost.

I agree that PEP 440 is more appropriate, but I think we must to be consistent here. We should either:

* Follow existing convention which is SemVer. I guess we will need to use the [pre-release form](https://semver.org/#spec-item-9). Something like:`2.5.1-dev.20190606`
* Or change the existing convention to PEP 440. 

---

Digging deeper into this: it looks like we only say that we use SemVer in the docs, but we actually don't follow version formats. For example, `2.5rc1` I believe is not a SemVer format (SemVer would be `2.5-rc.1`). I guess it's because we copy-pasted Django's release docs and `django.utils.version` :)

Our versioning actually look more like PEP 440. Does it make sense to update misleading docs and comments in [`wagtail.utils.version`](https://github.com/wagtail/wagtail/blob/669c1af4a0f4634395a63fd8f4d6a8f42f539730/wagtail/utils/version.py) while we are here?

Reply:
I don't think we strictly follow the format of the semver spec. That paragraph is just trying to say that we avoid breaking stuff in patch releases (or at least try to).

Since this is a python package, this version will need to be understood by setuptools, pip and PyPI. All of these expect PEP 440 formatted versions so I think that is what we should go with.

Reply:
Just to be clear, my issue is not with whether `dev` follows the relevant spec: the problem is that the `wagtail.utils.version.get_version` function fails with a `KeyError` when passed a string other than 'final', 'alpha', 'beta' or 'rc':

    >>> from wagtail.utils.version import get_version
    >>> get_version([2, 6, 0, 'dev', 1])
    Traceback (most recent call last):
      File "<console>", line 1, in <module>
      File "/vagrant/wagtail/wagtail/utils/version.py", line 17, in get_version
        sub = mapping[version[3]] + str(version[4])
    KeyError: 'dev'

https://github.com/wagtail/wagtail/blob/4a1f26778f1f48dc485eaa000b761dc995e40102/wagtail/utils/version.py#L15-L16

Reply:
> Since this is a python package, this version will need to be understood by setuptools, pip and PyPI. All of these expect PEP 440 formatted versions so I think that is what we should go with.

I agree. And we currently seem to use the right format. But the documentation and comments in `wagtail.utils.version` are currently misleading. I think we should just update these bits to avoid further confusion.

Reply:
@gasman should be fixed now